sql/_30_banana_pie_qa/issue_20447_disk_size/cases/20447_01.sql

```diff
@@ -0,0 +1,78 @@
+--+ holdcas on;
+
+drop table if exists t;
+
+create table t(s1 varchar,s2 char(500));
+
+insert into t values(repeat('a',200),repeat('b',200));
+
+insert into t values(repeat('b',300),repeat('c',300));
+
+select disk_size() from t;
+
+select disk_size(s1),disk_size(s2) from t order by 1;
+
+select t into :a from t order by s1 desc limit 1;
+
+select disk_size(:a.s1),:a.s1;
+
+create index idx_t on t(s1,s2) where disk_size(s2) > 10;
+
+drop t;
+
+
+-- disk_size with operation +,-,*,/,>,<...
+drop table if exists t1;
+
+create table t1(s1 varchar,i1 int,s2 char(200),s3 nchar(100),s4 nchar varying(500));
+
+insert into t1 values(null,1,null,null,null);
+
+insert into t1 values('',-1,'',n'',n'');
+
+insert into t1 values(repeat('a',300),0,repeat('a',300),repeat(n'a',300),repeat(n'a',300));
+
+select disk_size(s1),disk_size(i1),disk_size(s2),disk_size(s3),disk_size(s4) from t1;
+
+select disk_size(s1)+20 from t1;
+
+select disk_size(s1)-20 from t1;
+
+select disk_size(s1)*20 from t1;
+
+select disk_size(s1)/20 from t1;
+
+select s1 from t1 where disk_size(s1) > 24 order by s1;
+
+select s1 from t1 where disk_size(s1) >= 24 order by s1;
+
+select s1 from t1 where disk_size(s1) < 24 order by s1;
+
+select s1 from t1 where disk_size(s1) <= 24 order by s1;
+
+select s1 from t1 where disk_size(s1) <> 24 order by s1;
+
+select disk_size(abc) from t1;
+
+select disk_size(123) from t1;
+
+drop t1;
+
+
+-- prepare
+prepare st from 'select disk_size(?)';
+
+execute st using 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa';
+
+deallocate prepare st;
+
+
+-- session variable
+set @a=repeat('a',300);
+
+select @a,disk_size(@a);
+
+select @a,disk_size(cast(@a as char(100)));
+
+

```

Comment:
1. add more string functions in disk_size(), such as disk_size(substr()), disk_size(rpad()), disk_size(rtrim())
2. add some cases that using index, such as 
create index on t1(c1);
insert into t1 values(repeat('abc',100));
select disk_size(c1) from t1 where c1 like 'abc%'
3. add some cases to test more data types, such as float, decimal, set, enum